build.yaml

```diff
@@ -42,6 +42,13 @@ steps:
           yarn test --ci --coverage
       displayName: run unit tests
 
+    - task: PublishTestResults@2
+      inputs:
+          testResultsFiles: $(System.DefaultWorkingDirectory)/test-results/unit/junit.xml
+          testRunTitle: $(Agent.JobName)
+      condition: always()
```

Comment:
does this mean it will run even if the "build" task or the "run test" task failed, if so we might need to remove this condition

Reply:
yes. it should. if the test fails, we still want to publish the results.

Reply:
I mean the task fails, the build task, there'll be nothing to publish, right?

Reply:
yes. in that case, it won't publish  anything. check web repo. we have the same code there as well. If we are not publishing test results as part of running tests, we have to run the step always.